storage/ActiongraphImpl.cc

```diff
@@ -485,6 +486,28 @@ namespace storage
 
 
     void
+    Actiongraph::Impl::generate_compound_actions(const Actiongraph* actiongraph)
+    {
+	compound_actions.clear();
+
+	auto compound_actions = CompoundAction::Generator(actiongraph).generate();
+	for (auto action : compound_actions)
+	    Impl::compound_actions.push_back(shared_ptr<CompoundAction>(action));
+    }
+
+
+    vector<const CompoundAction*>
+    Actiongraph::Impl::get_compound_actions() const
+    {
+	vector<const CompoundAction*> actions;
+	for (auto action : compound_actions)
```

Comment:
The variable names are a bit inconsistent: E.g. ```action``` should be named ```compound_action```. The return value could be named ```ret``` to solve the name collision.